api/grpcserver/activation_service.go

```diff
@@ -30,11 +32,24 @@ func (s *activationService) RegisterService(server *Server) {
 
 // Get implements v1.ActivationServiceServer.
 func (s *activationService) Get(ctx context.Context, request *pb.GetRequest) (*pb.GetResponse, error) {
-	if l := len(request.Id); l != types.ATXIDSize {
+	var atxId types.ATXID
+	l := len(request.Id)
+	if l == 0 {
```

Comment:
i don't see why we should have such hacks

Reply:
`pb.GetRequest` is a generated type. I'm not sure its easy to add methods to it.

Reply:
yes method for the service :) Highest() or Commitment()

Reply:
thanks. added PR in api repository https://github.com/spacemeshos/api/pull/248
and updated the code here as well.